scripts/nix-profile-daemon.sh.in

```diff
@@ -0,0 +1,54 @@
+# Only execute this file once per shell.
+if [ -n "$__ETC_PROFILE_NIX_SOURCED" ]; then return; fi
+__ETC_PROFILE_NIX_SOURCED=1
+
+# Set up secure multi-user builds: non-root users build through the
+# Nix daemon.
+if [ "$USER" != root -o ! -w @localstatedir@/nix/db ]; then
+    export NIX_REMOTE=daemon
+fi
+
+export NIX_USER_PROFILE_DIR="@localstatedir@/nix/profiles/per-user/$USER"
+export NIX_PROFILES="@localstatedir@/nix/profiles/default $HOME/.nix-profile"
+
+# Set up the per-user profile.
+mkdir -m 0755 -p $NIX_USER_PROFILE_DIR
+if test "$(stat -f '%u' $NIX_USER_PROFILE_DIR)" != "$(id -u)"; then
```

Comment:
I seem to remember from the last attempt to have a multi-user-capable `nix-profile` that `stat` had issues on some platforms. But maybe we don't need to care if we're only going to use this on Linux and macOS.

Reply:
I tested `! test -O "$NIX_USER_PROFILE_DIR"` and it does fix the problems discussed in issue #1496, including `nix-shell -p bash --run bash` and nested nix-shell invocations.  It's also simpler and shorter.

However, it looks like it's [not in the POSIX standard](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/test.html).